Katello/katello
# Fixes #37383 - Add OSTree Include/Exclude Refs options

#### What are the changes introduced in this pull request?

Since Pulp OSTree plugin version 2.0.0 it's possible to use `include_refs` and `exclude_refs` parameters,
these are especially handy when you want to clone Fedora OSTree (Atomic Desktop) variants.

This PR adds theses 2 `generic_remote_option` parameters.

#### Considerations taken when implementing this change?

It was also necessary to do a db migration for already existing repos,
due to my very limited knowledge how to do that, my migration might look a bit bare minimum,
I would be very happy about a suggestion how to do that in a better way if there is one 🙂 

~~I also noticed that pulp takes the values on creation, but somehow in my test environment it only updated in the Katello db but not in pulp, couldn't find any errors or other code parts which would indicate that to happen. (will put in more time to find what's going on, but maybe it's already apparent to you what it is)~~

I saw that there is the possibility that OSTree repos get added by a subscription, these shouldn't have anything in the `generic_remote_options` column so the migration should also be fine there.

#### What are the testing steps for this pull request?

- Setup environment with OSTree enabled
- Add OSTree repo (I recommend using a small one, Fedora repos still take a long time, 9h, with the include filter)
- Parameters should be carried over into the pulp config
- And the sync should work as expected with the limited head ref filter

Looks like this when set up:
![image](https://github.com/Katello/katello/assets/42647570/82f28e4b-c1de-48b1-99a8-94222b76511f)
